akka-actor/src/main/scala/akka/serialization/Serialization.scala

```diff
@@ -247,13 +247,22 @@ class Serialization(val system: ExtendedActorSystem) extends Extension {
     Map(NullSerializer.identifier → NullSerializer) ++ serializers map { case (_, v) ⇒ (v.identifier, v) }
 
   private val isJavaSerializationWarningEnabled = settings.config.getBoolean("akka.actor.warn-about-java-serializer-usage")
+  private val suppressWarningOnNoVerification = settings.config.getBoolean("akka.actor.suppress-warning-on-no-verification")
 
   private def shouldWarnAboutJavaSerializer(serializedClass: Class[_], serializer: Serializer) = {
+
+    def shouldWarnOnNonSerializationVerification(serializedClass: Class[_]) = {
+      if (suppressWarningOnNoVerification)
+        !serializedClass.isInstanceOf[NoSerializationVerificationNeeded]
+      else
+        true
+    }
```

Comment:
I was scratching my head on how to make this condition easy to read... hopefully the latest commit is bit clearer to people, but please let me know if otherwise .... :|